driver-core/src/main/com/mongodb/internal/connection/SocksSocket.java

```diff
@@ -0,0 +1,783 @@
+
+        public byte getCommandNumber() {
+            return value;
+        }
+    }
+
+    private enum SocksAuthenticationMethod {
+        NO_AUTH(0x00),
+        USERNAME_PASSWORD(0x02);
+
+        private final byte methodNumber;
+
+        SocksAuthenticationMethod(final int methodNumber) {
+            this.methodNumber = (byte) methodNumber;
+        }
+
+        public byte getMethodNumber() {
+            return methodNumber;
+        }
+    }
+
+    enum AddressType {
+        IP_V4(0x01, 4),
+        IP_V6(0x04, 16),
+        DOMAIN_NAME(0x03, -1) {
+            public byte getLength() {
+                throw fail();
+            }
+        };
+
+        private final byte length;
+        private final byte addressTypeNumber;
+
+        AddressType(final int addressTypeNumber, final int length) {
+            this.addressTypeNumber = (byte) addressTypeNumber;
+            this.length = (byte) length;
+        }
+
+        static AddressType of(final byte signedAddressType) throws ConnectException {
+            int addressTypeNumber = Byte.toUnsignedInt(signedAddressType);
+            for (AddressType addressType : AddressType.values()) {
+                if (addressTypeNumber == addressType.getAddressTypeNumber()) {
+                    return addressType;
+                }
+            }
+            throw new ConnectException("Reply from SOCKS proxy server contains wrong address type"
+                    + " Address type: " + addressTypeNumber);
+        }
+
+        byte getLength() {
+            return length;
+        }
+
+        byte getAddressTypeNumber() {
+            return addressTypeNumber;
+        }
+
+    }
+
+    enum ServerReply {
+        REPLY_SUCCEEDED(0x00, "Succeeded"),
+        GENERAL_FAILURE(0x01, "General SOCKS5 server failure"),
+        NOT_ALLOWED(0x02, "Proxy server general failure"),
+        NET_UNREACHABLE(0x03, "Connection not allowed by ruleset"),
+        HOST_UNREACHABLE(0x04, "Network is unreachable"),
+        CONN_REFUSED(0x05, "Host is unreachable"),
+        TTL_EXPIRED(0x06, "Connection has been refused"),
+        CMD_NOT_SUPPORTED(0x07, "TTL expired"),
+        ADDR_TYPE_NOT_SUP(0x08, "Address type not supported");
+
+        private final int replyNumber;
+        private final String message;
+
+        ServerReply(final int replyNumber, final String message) {
+            this.replyNumber = replyNumber;
+            this.message = message;
+        }
+
+        static ServerReply of(final byte byteStatus) throws ConnectException {
+            int status = Byte.toUnsignedInt(byteStatus);
+            for (ServerReply serverReply : ServerReply.values()) {
+                if (status == serverReply.getReplyNumber()) {
+                    return serverReply;
+                }
+            }
+
+            throw new ConnectException("Unknown reply field. Reply field: " + status);
+        }
+
+        public int getReplyNumber() {
+            return replyNumber;
+        }
```

Comment:
This method is not needed because the `replyNumber` can be used instead in `ServerReply.of`. Alternatively, it can be made `private`, but there is really no reason.